internal/backend/temporalclient/client_test.go

```diff
@@ -0,0 +1,47 @@
+package temporalclient
+
+import (
+	"context"
+	"io"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/testsuite"
+	"go.uber.org/zap/zaptest"
+)
+
+func TestStartDevServer(t *testing.T) {
+	dataDir := t.TempDir()
+	t.Setenv("XDG_DATA_HOME", dataDir)
+
+	ctx := context.Background()
+	logger := zaptest.NewLogger(t)
+	cfg := &Config{
+		DevServer: testsuite.DevServerOptions{},
+	}
+	opts := client.Options{}
+
+	c := &impl{
+		done: make(chan struct{}),
+		z:    logger,
+	}
+	err := c.startDevServer(ctx, cfg, opts)
+	require.NoError(t, err)
+	defer c.Stop(ctx)
+
+	require.NotNil(t, c.client)
+	require.NotNil(t, c.srv)
+	require.NotNil(t, c.srvLog)
+
+	time.Sleep(100 * time.Millisecond) // Let the server emit some logs
+
+	file, err := os.Open(c.srvLog.Name())
+	require.NoError(t, err)
+	defer file.Close()
+	data, err := io.ReadAll(file)
+	require.NoError(t, err)
+	require.NotEmpty(t, data)
```

Comment:
temporal log file could exists before the test and then this test will pass anyway.
I suggest either to make temporal log file name configurable, then you could ensure its uniqueness before the test
or search for specific message in data with time stamp or some random identifier that you could recognize

Reply:
Added the following at the start of the test to get a unique directory:

```
	dataDir := t.TempDir()
	t.Setenv("XDG_DATA_HOME", dataDir)
```